File: SkyModel.Core/Models/MeshData.cs
namespace SkyModel.Core.Models;

public sealed class MeshData
{
    public required string Name { get; init; }
    public required string SourcePath { get; init; }
    public required List<(float X, float Y, float Z)> Vertices { get; init; }
    public required List<(byte B0, byte B1, byte B2, byte B3)> PackedVertexAttrs { get; init; }
    public required List<(float U, float V)> Uv0 { get; init; }
    public required List<(int A, int B, int C)> Indices { get; init; }
    public required List<List<BoneWeight>> BoneWeights { get; init; }
    public required List<SkeletonBone>? EmbeddedSkeleton { get; init; }
    public required int Version { get; init; }
    public required bool IsAnimated { get; init; }
}

File: SkyModelViewerAIDE/app/src/main/java/com/sky/modelviewer/model/MeshData.java
package com.sky.modelviewer.model;

import java.util.List;

public class MeshData {
    public final String name;
    public final String sourcePath;
    public final List<float[]> vertices;
    public final List<byte[]> packedVertexAttrs;
    public final List<float[]> uv0;
    public final List<int[]> indices;
    public final List<List<BoneWeight>> boneWeights;
    public final List<SkeletonBone> embeddedSkeleton;
    public final int version;
    public final boolean isAnimated;

    public MeshData(String name, String sourcePath, List<float[]> vertices,
                    List<byte[]> packedVertexAttrs, List<float[]> uv0,
                    List<int[]> indices, List<List<BoneWeight>> boneWeights,
                    List<SkeletonBone> embeddedSkeleton, int version, boolean isAnimated) {
        this.name = name;
        this.sourcePath = sourcePath;
        this.vertices = vertices;
        this.packedVertexAttrs = packedVertexAttrs;
        this.uv0 = uv0;
        this.indices = indices;
        this.boneWeights = boneWeights;
        this.embeddedSkeleton = embeddedSkeleton;
        this.version = version;
        this.isAnimated = isAnimated;
    }
}
